doc/md/5-actors-async.md

```diff
@@ -0,0 +1,316 @@
+
+For example, the implementation of `bump()` above is guaranteed to increment and read the value of `count`, in one atomic step. The following alternative implementation does not have the same semantics and allows another client of the actor to interfere with its operation.
+
+``` motoko no-repl
+  public shared func bump() : async Nat {
+    await inc();
+    await read();
+  };
+```
+
+Each `await` suspends execution, allowing an interloper to change the state of the actor. By design, the explicit `await`s make the potential points of interference clear to the reader.
+
+:::
+
+## Async actors
+
+In Motoko, each communicating component is an actor, encapsulating its own state and behavior. Here's a simple three-line example that demonstrates basic actor usage:
+
+```motoko no-repl
+let result1 = service1.computeAnswer(params);
+let result2 = service2.computeAnswer(params);
+finalStep(await result1, await result2)
+```
+
+This program’s behavior can be summarized as:
+
+1.  The program makes two requests (lines 1 and 2) to two distinct services, each implemented as a Motoko actor or canister smart contract implemented in some other language.
+
+2.  The program waits for each result to be ready (line 3) using the keyword `await` on each result value.
+
+3.  The program uses both results in the final step (line 3) by calling the `finalStep` function.
+
+Services **interleave** their execution to reduce latency instead of waiting for each other. Without language support, this kind of interleaving quickly becomes complex and hard to manage.
+
+Even with just a single async call, Motoko’s abstractions help keep the code clear. By using `await`, the programmer tells the compiler where interleaving can happen, avoiding the need to restructure logic to fit the system’s message-passing loop.
+
+In other languages without these features, developers often need to use advanced patterns like callbacks and event handlers. This low-level, systems-style programming can be powerful but is error-prone, as it breaks high-level logic into scattered events and shared state.
+
+### Example
+
+To demonstrate how asynchronous actors work, consider the following example.
+
+Customers place orders at a pizza restaurant, but the chef can only make one pizza at a time. Orders are taken **[asynchronously](https://internetcomputer.org/docs/motoko/fundamentals/actors-async#async--await)**, meaning customers do not have to wait for previous orders to be completed before placing their own. However, each pizza is prepared sequentially. This is representative of an asynchronous actor.
+
+```motoko no-repl
+import Array "mo:base/Array";
```

Comment:
It would be better use a Deque (Queue) pushing new order to the ends and popping the next order to make from the front.